src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSPackageRestoreManager.cs

```diff
@@ -29,32 +31,42 @@ internal sealed class VSPackageRestoreManager : PackageRestoreManager
 
             SolutionManager = solutionManager;
             SolutionManager.NuGetProjectAdded += OnNuGetProjectAdded;
-            SolutionManager.SolutionOpened += OnSolutionOpenedOrClosed;
-            SolutionManager.SolutionClosed += OnSolutionOpenedOrClosed;
+            SolutionManager.SolutionOpened += OnSolutionOpened;
+            SolutionManager.SolutionClosed += OnSolutionClosed;
         }
 
-        private void OnSolutionOpenedOrClosed(object sender, EventArgs e)
+        private void OnSolutionOpened(object sender, EventArgs e)
         {
             // This is a solution event. Should be on the UI thread
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
-                {
-                    // We need to do the check even on Solution Closed because, let's say if the yellow Update bar
-                    // is showing and the user closes the solution; in that case, we want to hide the Update bar.
-                    var solutionDirectory = SolutionManager.SolutionDirectory;
-                    await RaisePackagesMissingEventForSolutionAsync(solutionDirectory, CancellationToken.None);
-                });
+            var solutionDirectory = SolutionManager.SolutionDirectory;
+
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
```

Comment:
Both here and below consider adding `FileAndForget(…)`.  See [this](https://github.com/microsoft/vs-threading/blob/b3850f3e67f6fb819ed4cc3e38ad5f1ab023c546/doc/cookbook_vs.md#task-returning-fire-and-forget-methods) and [this](https://github.com/NuGet/NuGet.Client/blob/1a0552a3066b4532dfdaf9b18faba99b86592e04/src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs#L751).

Reply:
Done.